cmd/kube-controller-manager/app/names/controller_names.go

```diff
@@ -0,0 +1,108 @@
+	ReplicationControllerControllerName              = "replicationcontroller-controller"
+	PodGarbageCollectorControllerName                = "pod-garbage-collector-controller"
+	ResourceQuotaControllerName                      = "resourcequota-controller"
+	NamespaceControllerName                          = "namespace-controller"
+	ServiceAccountControllerName                     = "serviceaccount-controller"
+	GarbageCollectorControllerName                   = "garbage-collector-controller"
+	DaemonSetControllerName                          = "daemonset-controller"
+	JobControllerName                                = "job-controller"
+	DeploymentControllerName                         = "deployment-controller"
+	ReplicaSetControllerName                         = "replicaset-controller"
+	HorizontalPodAutoscalerControllerName            = "horizontal-pod-autoscaler-controller"
+	DisruptionControllerName                         = "disruption-controller"
+	StatefulSetControllerName                        = "statefulset-controller"
+	CronJobControllerName                            = "cronjob-controller"
+	CertificateSigningRequestSigningControllerName   = "certificatesigningrequest-signing-controller"
+	CertificateSigningRequestApprovingControllerName = "certificatesigningrequest-approving-controller"
+	CertificateSigningRequestCleanerControllerName   = "certificatesigningrequest-cleaner-controller"
+	TTLControllerName                                = "ttl-controller"
+	BootstrapSignerControllerName                    = "bootstrap-signer-controller"
+	TokenCleanerControllerName                       = "token-cleaner-controller"
+	NodeIpamControllerName                           = "node-ipam-controller"
+	NodeLifecycleControllerName                      = "node-lifecycle-controller"
+	PersistentVolumeBinderControllerName             = "persistentvolume-binder-controller"
+	AttachDetachControllerName                       = "attach-detach-controller"
+	PersistentVolumeExpanderControllerName           = "persistentvolume-expander-controller"
+	ClusterRoleAggregationControllerName             = "clusterrole-aggregation-controller"
+	PersistentVolumeClaimProtectionControllerName    = "persistentvolumeclaim-protection-controller"
+	PersistentVolumeProtectionControllerName         = "persistentvolume-protection-controller"
+	TTLAfterFinishedControllerName                   = "ttl-after-finished-controller"
+	RootCACertPublisherControllerName                = "root-ca-cert-publisher-controller"
+	EphemeralVolumeControllerName                    = "ephemeral-volume-controller"
+	StorageVersionGarbageCollectorControllerName     = "storageversion-garbage-collector-controller"
+	ResourceClaimControllerName                      = "resourceclaim-controller"
+)
+
+// KCMControllerAliases is a mapping of aliases to canonical controller names
+var KCMControllerAliases = map[string]string{
```

Comment:
making a global exported map means it is subject to mutation by other packages, which I don't think we want... consider making a way to access this without allowing modifying the original

Reply:
yes that would not be nice - using functions  now instead